Cisco IOS Preset

grp-db opened the pull request:
Preset for [Cisco](https://www.cisco.com/c/en/us/td/docs/routers/access/wireless/software/guide/SysMsgLogging.html):[IOS](https://www.cisco.com/c/en/us/td/docs/routers/access/wireless/software/guide/SysMsgLogging.html#wp1054470)

grp-db committed:
Cisco IOS Preset

grp-db committed:
Cisco IOS Preset

1. add metadata version

grp-db committed:
Cisco IOS Preset

1. add metadata version to all 4 ocsf schemas

grp-db committed:
Cisco IOS Preset

1. add transparent icon
2. fix header tags
3. add meta fields

alexott commented on presets/cisco/ios/preset.yaml:
```diff
@@ -0,0 +1,451 @@
+      - CAST('cisco' AS STRING) AS source
+      - CAST('ios' AS STRING) AS sourcetype
+      - CURRENT_TIMESTAMP() as processed_time
+silver:
```
Do we really need to have multiple silver tables here?

grp-db replied:
I noticed a lot of the existing presets have separate silver per event class (i.e. okta).

alexott reviewed (commented):
Plus the same comments as in #71 and #72

hannes-antimatter reviewed (commented):
Apologies for the delay - highly appreciate your contribution, thanks! If you could answer the few questions below I'm going to approve the PR since it generally looks good.

Please note that we cannot merge it at this point as we are in a transition of our backend. We'll come back to this once that has been resolved.

Here are some notes / recommendations:
As Alex mentioned, it is recommended to keep the number of silver tables as small as possible. There is no clear guidance currently and depends on the specific dataset complexity.
A good indication is that you essentially move your filter criterion for each silver tables directly into the appropriate gold table (without any complex transformations).


Few question:
- on how many rows have you tested the preset ?
- are you able to share some slice of the example data with us ? (looking to use them for automated testing in the future etc)
- there are no schemaHints - can you comment on how "robust" the auto-type inference works on that data? Background of that question here is that we generally want presets to "always work". Since it is text format we cannot use autoloader's single variant ingest. But I wonder if it could be made more robust against unexpected type conversions/variations in the data.

grp-db committed:
Cisco IOS Preset

convert back to single Silver (update readme, preset)

grp-db commented:
Thanks! I switched back to a single silver and added schema hints for value string. I shared the preset notebooks with Allen to review the data transformations on the sample logs.

hannes-antimatter reviewed (approved):
Thanks - looks good!

As mentioned above: please do not merge for now ! We'll get back once we finished our backend transition.

alexott commented:
Moved to the content pack
